### PR TITLE
Feat/indent lists

### PR DESCRIPTION
**Pull request by maimonav**
Click below to open examples:
rich-content: https://rich-content-indentLists.surge.sh/
rich-content-storybook: https://rich-content-storybook-indentLists.surge.sh/

### CODE REVIEW

File: packages/editor/web/src/RichContentEditor/handleKeyCommand.js [hunk 1]
```diff
@@ -1,10 +1,18 @@
 import { COMMANDS, mergeBlockData, RichUtils } from 'wix-rich-content-editor-common';
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/handleKeyCommand.js:1]: are these changes needed?

File: packages/editor/web/statics/styles/rich-content-editor.scss [hunk 2]
```diff
@@ -133,14 +133,6 @@
   text-indent: 5%;
 }
 
-.orderedList {
-  composes: elementSpacing;
```
Comment (shaulgo) [inline packages/editor/web/statics/styles/rich-content-editor.scss:137]: what does this do? Why remove it?

File: packages/editor/web/src/RichContentEditor/RichContentEditor.jsx [hunk 3]
```diff
@@ -245,27 +248,35 @@ class RichContentEditor extends Component {
...
     commands: [
       ...this.pluginKeyBindings.commands,
       {
-        command: 'tab',
+        command: COMMANDS.TAB,
         modifiers: [],
         key: 'Tab',
       },
+      {
+        command: COMMANDS.SHIFT_TAB,
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/RichContentEditor.jsx:271]: I think that this is the only change that is needed

File: packages/plugin-code-block/web/src/createCodeBlockPlugin.js [hunk 4]
```diff
@@ -1,31 +1,18 @@
...
-        return CodeUtils.getKeyBinding(event);
-      }
-    }
-  },
-
   handleKeyCommand: (command, editorState, timestamp, { setEditorState }) => {
     if (CodeUtils.hasSelectionInBlock(editorState)) {
       let newState;
-      if (command === 'tab') {
+      if (command === COMMANDS.TAB) {
```
Comment (shaulgo) [inline packages/plugin-code-block/web/src/createCodeBlockPlugin.js:11]: very nice

File: packages/plugin-code-block/web/src/createCodeBlockPlugin.js [hunk 5]
```diff
@@ -1,31 +1,18 @@
...
         const mockEvent = { preventDefault: () => {} };
         newState = CodeUtils.onTab(mockEvent, editorState);
-      } else {
-        newState = CodeUtils.handleKeyCommand(editorState, command);
+      } else if (command === COMMANDS.SHIFT_TAB) {
+        newState = CodeUtils.handleKeyCommand(editorState, 'backspace');
```
Comment (shaulgo) [inline packages/plugin-code-block/web/src/createCodeBlockPlugin.js:15]: add the deleted comment. Maybe even add a function called handShiftTab. This way it will be easy to see the comment. Otherwise it will be hard to find this. 
handShiftTab(editorState) {
// since backspace removes tabs in CodeUtils	
// https://github.com/SamyPesse/draft-js-code/blob/9783c0f6bbedda6b7089712f9c657a72fdae636d/lib/handleKeyCommand.js#L11
return newState = CodeUtils.handleKeyCommand(editorState, 'backspace')
}

File: packages/plugin-code-block/web/src/createCodeBlockPlugin.js (hunk 4 above)
Comment (shaulgo) [reply packages/plugin-code-block/web/src/createCodeBlockPlugin.js:11]: Reminder for me about split prs.

File: packages/editor/web/statics/styles/rich-content-editor.scss (hunk 2 above)
Comment (maimonav) [reply packages/editor/web/statics/styles/rich-content-editor.scss:137]: It used to be class name style for styles, now i'm using draft.

File: packages/plugin-code-block/web/src/createCodeBlockPlugin.js (hunk 4 above)
Comment (maimonav) [reply packages/plugin-code-block/web/src/createCodeBlockPlugin.js:11]: Thanks.

File: packages/editor/web/src/RichContentEditor/handleKeyCommand.js (hunk 1 above)
Comment (maimonav) [reply packages/editor/web/src/RichContentEditor/handleKeyCommand.js:1]: Yes, here i use the mechanism of draft with RichUtils.onTab,
While not effecting other on tab events such in code block for example.

File: package.json [hunk 6]
```diff
@@ -129,5 +130,8 @@
   },
   "publishConfig": {
     "registry": "https://registry.npmjs.org/"
+  },
+  "dependencies": {
+    "draft-js-list-plugin": "^1.0.2"
```
[the remark this answers is not in the document]
Comment (maimonav) [reply package.json:135]: Yes, this will be added to dependenciesDev in Editor.

File: packages/editor/web/src/RichContentEditor/RichContentEditor.jsx (hunk 3 above)
Comment (maimonav) [reply packages/editor/web/src/RichContentEditor/RichContentEditor.jsx:271]: Since we added support for tab & shift-tab in keybindings,
we need this changes below.

File: packages/plugin-code-block/web/src/createCodeBlockPlugin.js (hunk 5 above)
Comment (shaulgo) [reply packages/plugin-code-block/web/src/createCodeBlockPlugin.js]: @maimonav

Comment (shaulgo) [reply packages/plugin-code-block/web/src/createCodeBlockPlugin.js]: @maimonav

Comment (maimonav) [reply packages/plugin-code-block/web/src/createCodeBlockPlugin.js]: I did it, this is outdated.
If you enter to files changed you'll see it.

File: packages/viewer/web/src/List.jsx [hunk 7]
```diff
@@ -1,18 +1,38 @@
...
+  const direction = rtl ? 'RTL' : 'LTR';
+  let className = draftClassNames(listType, depth, direction);
+  if (i === 0 || depth > prevDepth) {
+    className += ` ${draftPublic}-reset`;
+  }
+  return className;
+};
+
+const getBlockDepth = (contentState, key) =>
+  contentState.blocks.filter(block => block.key === key)[0].depth;
```
Comment (shaulgo) [inline packages/viewer/web/src/List.jsx:21]:
```suggestion
  contentState.blocks.find(block => block.key === key).depth;
```

File: packages/viewer/web/src/List.jsx [hunk 8]
```diff
@@ -1,18 +1,38 @@
...
+const draftClassNames = (listType, depth, textDirection) =>
+  `${draftPublic}-${listType}ListItem
+   ${draftPublic}-depth${depth}
+   ${draftPublic}-list${textDirection}`;
+
+const getBlockClassName = (i, dataEntry, textDirection, listType, prevDepth, depth) => {
+  const rtl = textDirection === 'rtl' || dataEntry.textDirection === 'rtl';
+  const direction = rtl ? 'RTL' : 'LTR';
+  let className = draftClassNames(listType, depth, direction);
+  if (i === 0 || depth > prevDepth) {
```
Comment (shaulgo) [inline packages/viewer/web/src/List.jsx:14]: isNewList

File: packages/editor/web/src/RichContentEditor/createPlugins.jsx [hunk 9]
```diff
@@ -15,7 +15,7 @@ const createPlugins = ({ plugins, context }) => {
     isMobile: context.isMobile,
   });
 
-  const listPlugin = createListPlugin({ olRegex: /1\./ });
+  const listPlugin = createListPlugin({ olRegex: /1\./, allowNestedLists: false });
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/createPlugins.jsx:18]: what does this do?

Comment (maimonav) [reply packages/editor/web/src/RichContentEditor/createPlugins.jsx:18]: In general, this disable nested lists.
But for us, it disables unwanted behaviour of keyBindings.

File: packages/viewer/web/src/RichContentViewer.jsx [hunk 10]
```diff
@@ -7,19 +7,19 @@ import {
   normalizeInitialState,
   getLangDir,
 } from 'wix-rich-content-common';
+import 'wix-rich-content-editor/dist/statics/styles/draftDefault.rtlignore.scss';
```
Comment (shaulgo) [inline packages/viewer/web/src/RichContentViewer.jsx:10]: needs to move into wix-rich-content-common. We don't import from the editor into the viewer.

Comment (maimonav) [reply packages/viewer/web/src/RichContentViewer.jsx:10]: ok sure.
